packages/cspell/src/lint.ts

```diff
@@ -0,0 +1,288 @@
+
+        const configErrors = await countConfigErrors(configInfo);
+        if (configErrors) return runResult({ errors: configErrors });
+
+        // Get Exclusions from the config files.
+        const { root } = cfg;
+        const ignoreGlobs = normalizeGlobsToRoot(configInfo.config.ignorePaths || [], root, true).concat(excludeGlobs);
+        const globOptions = { root, cwd: root, ignore: ignoreGlobs };
+        const exclusionGlobs = extractGlobExcludesFromConfig(root, configInfo.source, configInfo.config).concat(
+            cfg.excludes
+        );
+        const files = filterFiles(await findFiles(fileGlobs, globOptions), exclusionGlobs);
+
+        return processFiles(fileLoader(files), configInfo, files.length);
+    }
+
+    function header(files: string[]) {
+        const formattedFiles = files.length > 100 ? files.slice(0, 100).concat(['...']) : files;
+
+        cfg.info(
+            `
+cspell;
+Date: ${new Date().toUTCString()}
+Options:
+    verbose:   ${yesNo(!!cfg.options.verbose)}
+    config:    ${cfg.configFile || 'default'}
+    exclude:   ${extractPatterns(cfg.excludes)
+        .map((a) => a.glob.glob)
+        .join('\n             ')}
+    files:     ${formattedFiles}
+    wordsOnly: ${yesNo(!!cfg.options.wordsOnly)}
+    unique:    ${yesNo(!!cfg.options.unique)}
+`,
+            MessageTypes.Info
+        );
+    }
+
+    function isExcluded(filename: string, globs: GlobSrcInfo[]) {
+        const { root } = cfg;
+        const absFilename = path.resolve(root, filename);
+        for (const glob of globs) {
+            const m = glob.matcher.matchEx(absFilename);
+            if (m.matched) {
+                cfg.info(
+                    `Excluded File: ${path.relative(root, absFilename)}; Excluded by ${m.glob} from ${glob.source}`,
+                    MessageTypes.Info
+                );
+                return true;
+            }
+        }
+        return false;
+    }
+
+    function filterFiles(files: string[], excludeGlobs: GlobSrcInfo[]): string[] {
+        const excludeInfo = extractPatterns(excludeGlobs).map((g) => `Glob: ${g.glob.glob} from ${g.source}`);
+        cfg.info(`Exclusion Globs: \n    ${excludeInfo.join('\n    ')}\n`, MessageTypes.Info);
+        const result = files.filter(util.uniqueFn()).filter((filename) => !isExcluded(filename, excludeGlobs));
+        return result;
+    }
+}
+
+function extractContext(tdo: cspell.TextDocumentOffset, contextRange: number): cspell.TextOffset {
+    const { line, offset } = tdo;
+    const textOffsetInLine = offset - line.offset;
+    let left = Math.max(textOffsetInLine - contextRange, 0);
+    let right = Math.min(line.text.length, textOffsetInLine + contextRange + tdo.text.length);
+    const lineText = line.text;
+
+    const isLetter = /^[a-z]$/i;
+    const isSpace = /^\s$/;
+
+    for (let n = contextRange / 2; n > 0 && left > 0 && isLetter.test(lineText[left - 1]); n--, left--) {
```

Comment:
think it reads a little better with `isLetter.test` part inside the loop

Reply:
I can see that.